helper_dart/lib/src/facet_list.dart

```diff
@@ -281,6 +296,15 @@ class _FacetList with DisposableMixin implements FacetList {
     });
   }
 
+  void _trackClickIfNeeded(String selection) {
+    _selections.first.then((selections) {
```

Comment:
Yes, `_trackClickIfNeeded` called before the filter state update, so we check the current state, if `_selections` contains the `selection` value, no event will be sent since that means it's already checked.